lib/Runtime/Math/WasmMath.inl

```diff
@@ -18,8 +124,77 @@ WasmMath::Ctz(int value)
     return 32;
 }
 
-inline int
-WasmMath::Eqz(int value)
+template<> 
+inline int64 WasmMath::Ctz(int64 value)
+{
+    DWORD index;
+#if TARGET_64
+    if (_BitScanForward64(&index, value))
+    {
+        return index;
+    }
+#else
+    if (_BitScanForward(&index, (int32)value))
+    {
+        return index;
+    }
+    if (_BitScanForward(&index, (int32)(value >> 32)))
+    {
+        return index + 32;
+    }
+#endif
+    return 64;
+}
+
+template<> 
+inline int64 WasmMath::Clz(int64 value)
+{
+    DWORD index;
+#if TARGET_64
+    if (_BitScanReverse64(&index, value))
+    {
+        return 63 - index;
+    }
+#else
+    if (_BitScanReverse(&index, (int32)(value >> 32)))
+    {
+        return 31 - index;
+    }
+    if (_BitScanReverse(&index, (int32)value))
+    {
+        return 63 - index;
+    }
+#endif
+    return 64;
+}
+
+template<> 
+inline int WasmMath::PopCnt(int value)
+{
+#if defined(_M_IX86) || defined(_M_X64)
+    return __popcnt(value);
+#else
+    AssertMsg(false, "WasmMath::PopCnt<int> not implemented for this architecture");
+    return 0;
+#endif
+}
+
+template<> 
+inline int64 WasmMath::PopCnt(int64 value)
+{
+#if _M_X64
+    return __popcnt64(value);
```

Comment:
popcnt is only available on very new hardware, can we do dynamic check and if pnpcnt isn't available do the slow algorithm for it?